downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -358,6 +386,23 @@ enum class VcsHost(
 
     protected abstract fun toVcsInfoInternal(projectUrl: URI): VcsInfo
 
+    /**
+     * Return the download URL to an archive generated for the referenced [vcsInfo], or null if no download URL can be
+     * determined.
+     */
+    fun toArchiveDownloadUrl(vcsInfo: VcsInfo): String? {
```

Comment:
I do not understand this duplication of the function from the companion object. Is this intended?

Reply:
Yes, it's intended. That's the pattern we currently follow in this class, see also e.g. the `toPermalink()` function, so that you can call these functions in case you already know the VcsHost, and also in case you don't already know the VcsHost (and want to get it auto-detected).

Reply:
Thank you for explaining.